Automattic/woocommerce-payments
# Add payment icons to multiple UPE

Fixes #4666 

#### Changes proposed in this Pull Request


Add Payment Icons to UPE Payment Methods



This PR adds an icon attribute to the UPE and CC payment methods that will be displayed at checkout. I opted to add the icon URL to the payment methods themselves rather than at the abstract class and derive the image based on the method name as it seemed a lot clearer to define all of the payment method specific items within the payment method classes. This also allows more flexibility between icon file names and payment method names. I could easily be convinced to change the approach though.

The implementation also takes advantage of the WooCommerce built in icon handling by just setting the icon attribute on the gateways. The relevant WooCommerce code can be found in these two spots if you wanted to review how the icons are used in WC:

https://github.com/woocommerce/woocommerce/blob/1eb02b4742d685b596ef518affe44637bf9cfb26/plugins/woocommerce/includes/abstracts/abstract-wc-payment-gateway.php#L326

https://github.com/woocommerce/woocommerce/blob/1eb02b4742d685b596ef518affe44637bf9cfb26/plugins/woocommerce/templates/checkout/payment-method.php#L26

**I made these changes off of the multiple UPE POC branch** as that was the only way to display the icons for all payment methods at checkout. **Let me know if this should be rebased off of develop though before merging**. This current arrangement just makes testing easier.





#### Testing instructions





* Checkout the current branch which was branched off of the multiple UPE branch
* Change currency to EUR to allow the relevant payment methods in UPE
* Set up an account and enable UPE
* Enable all relevant payment methods in UPE
* As a customer add item to cart and go to checkout
* Should see icons next to the payment methods as pictured below

![UPE-payment-icons](https://user-images.githubusercontent.com/68524302/194627653-e00ab3bf-ad2e-415e-8d64-7a9a21e0439e.png)

* Disable UPE and check for CC icons still displayed

![CC-payment-icons](https://user-images.githubusercontent.com/68524302/194627757-db074207-9c48-4d00-bd7a-fce6154e746d.png)


-------------------


**Post merge**
